### ISSUE TITLE
Segmentation violation when adding git repo with chart containing selectors

### ISSUE THREAD

**Issue by dnoland1** [rancher/fleet#203]
1. On Rancher v2.5.3, add a one node cluster with all roles and label "production = true".
2. Go to Continuous Delivery. Add a GitRepo from YAML and use:
```yaml
​apiVersion: fleet.cattle.io/v1alpha1
kind: GitRepo
metadata:
  name: 'testrepo'
  namespace: fleet-default
spec:
  branch: main
  forceSyncGeneration: 5
  paths: []
  repo: >-
    https://github.com/waterfoul/testRepo.git
  targets:
    - clusterSelector:
        matchLabels:
          staging: 'true'
      name: staging
    - clusterSelector:
        matchLabels:
          production: 'true'
      name: production
type: fleet.cattle.io.gitrepo
```
Click into the repo, after about a minute, you'll see the error:
![image](https://user-images.githubusercontent.com/468462/103319326-50b0c580-49e6-11eb-94e5-6f5d044d13eb.png)

Logs from container:
```
# kubectl logs -n fleet-default testrepo-5faa5-65j8r fleet
KUBERNETES_SERVICE_PORT_HTTPS=443
KUBERNETES_SERVICE_PORT=443
HOSTNAME=testrepo-5faa5-65j8r
COMMIT=d730f7faadd0ec25eea696607b85694db599ddb2
PWD=/workspace/source
EVENT_TYPE=
HOME=/root
KUBERNETES_PORT_443_TCP=tcp://10.43.0.1:443
SHLVL=1
KUBERNETES_PORT_443_TCP_PROTO=tcp
KUBERNETES_PORT_443_TCP_ADDR=10.43.0.1
KUBERNETES_SERVICE_HOST=10.43.0.1
KUBERNETES_PORT=tcp://10.43.0.1:443
KUBERNETES_PORT_443_TCP_PORT=443
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
_=/usr/bin/env
fleet-default/testrepo
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x196251b]

goroutine 1 [running]:
github.com/rancher/fleet/pkg/bundle.addCharts(0xc0008d6d40, 0x1, 0x1, 0xc0008adf9c, 0x3, 0xc00000fa10, 0x1, 0x1, 0x1, 0xc0008d6d40, ...)
	/go/src/github.com/rancher/fleet/pkg/bundle/resources.go:129 +0x9b
github.com/rancher/fleet/pkg/bundle.readResources(0x27d8080, 0xc000334680, 0xc000914dd0, 0x22a1400, 0xc0008adf9c, 0x3, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/rancher/fleet/pkg/bundle/resources.go:49 +0x205
github.com/rancher/fleet/pkg/bundle.read(0x27d8080, 0xc000334680, 0xc00092a060, 0xc, 0xc0008adf9c, 0x3, 0x2773060, 0xc0008f4b10, 0xc00099f380, 0x1f17020, ...)
	/go/src/github.com/rancher/fleet/pkg/bundle/read.go:130 +0x357
github.com/rancher/fleet/pkg/bundle.Read(0x27d8080, 0xc000334680, 0xc00092a060, 0xc, 0xc0008adf9c, 0x3, 0x2777240, 0xc00000f9e8, 0xc00099f380, 0xc00092a040, ...)
	/go/src/github.com/rancher/fleet/pkg/bundle/read.go:73 +0x150
github.com/rancher/fleet/pkg/bundle.Open(0x27d8080, 0xc000334680, 0xc00092a060, 0xc, 0xc0008adf9c, 0x3, 0x0, 0x0, 0xc00099f380, 0x0, ...)
	/go/src/github.com/rancher/fleet/pkg/bundle/read.go:60 +0x245
github.com/rancher/fleet/modules/cli/apply.readBundle(0x27d8080, 0xc000334680, 0xc00092a060, 0xc, 0xc0008adf9c, 0x3, 0xc00099fb18, 0xc0008adfd0, 0xe, 0xc0008adfd4)
	/go/src/github.com/rancher/fleet/modules/cli/apply/apply.go:120 +0x29b
github.com/rancher/fleet/modules/cli/apply.Dir(0x27d8080, 0xc000334680, 0xc00055e1b0, 0x7ffc7399edee, 0x8, 0xc0008adf9c, 0x3, 0xc00099fb18, 0x0, 0x1)
	/go/src/github.com/rancher/fleet/modules/cli/apply/apply.go:142 +0xd9
github.com/rancher/fleet/modules/cli/apply.Apply.func1(0xc0008adf9c, 0x3, 0x27eba40, 0xc000914c30, 0x0, 0x0, 0x0, 0xc000914c30)
	/go/src/github.com/rancher/fleet/modules/cli/apply/apply.go:89 +0x1b0
path/filepath.walk(0xc0008adf9c, 0x3, 0x27eba40, 0xc000914c30, 0xc00099f988, 0x0, 0x0)
	/usr/local/go/src/path/filepath/path.go:364 +0xe3
path/filepath.walk(0x38bf901, 0x1, 0x27eba40, 0xc000914a90, 0xc00099f988, 0x0, 0x0)
	/usr/local/go/src/path/filepath/path.go:384 +0x2ff
path/filepath.walk(0x7ffc7399edf7, 0x1, 0x27eba40, 0xc000033450, 0xc00099f988, 0x0, 0x7ffc7399edf7)
	/usr/local/go/src/path/filepath/path.go:384 +0x2ff
path/filepath.Walk(0x7ffc7399edf7, 0x1, 0xc00065d988, 0x1, 0x1)
	/usr/local/go/src/path/filepath/path.go:406 +0xff
github.com/rancher/fleet/modules/cli/apply.Apply(0x27d8080, 0xc000334680, 0xc00055e1b0, 0x7ffc7399edee, 0x8, 0xc0004c6190, 0x1, 0xb, 0xc00099fb18, 0x7f90570576e0, ...)
	/go/src/github.com/rancher/fleet/modules/cli/apply/apply.go:79 +0x314
github.com/rancher/fleet/modules/cli/cmds.(*Apply).Run(0xc000031200, 0xc00074f600, 0xc0004c6180, 0x2, 0xc, 0x0, 0x0)
	/go/src/github.com/rancher/fleet/modules/cli/cmds/apply.go:85 +0x2c3
github.com/rancher/wrangler-cli.bind.func1(0xc00074f600, 0xc0004c6180, 0x2, 0xc, 0x0, 0x0)
	/go/pkg/mod/github.com/rancher/wrangler-cli@v0.0.0-20200815040857-81c48cf8ab43/builder.go:249 +0x113
github.com/spf13/cobra.(*Command).execute(0xc00074f600, 0xc0004c60c0, 0xc, 0xc, 0xc00074f600, 0xc0004c60c0)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842 +0x453
github.com/spf13/cobra.(*Command).ExecuteC(0xc00074f340, 0xc000054060, 0x27d8080, 0xc000334680)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:880
github.com/rancher/wrangler-cli.Main(0xc00074f340)
	/go/pkg/mod/github.com/rancher/wrangler-cli@v0.0.0-20200815040857-81c48cf8ab43/builder.go:73 +0x6d
main.main()
	/go/src/github.com/rancher/fleet/main.go:19 +0x27
```

gz#14124

**Comment by jiaqiluo** [comment]
The bug is reproduced on Rancher:`v2.5.3` single-install.

The steps are the same as the issue provides. 

Results:

![screencapture-165-227-51-236-dashboard-c-c-d74bh-fleet-fleet-cattle-io-gitrepo-fleet-default-testrepo-2021-01-04-11_09_33](https://user-images.githubusercontent.com/6218999/103566196-cbfaf700-4e7e-11eb-8376-b1fd582f1c24.png)

**Comment by jiaqiluo** [comment]
The bug fix is validated on Rancher:v2.5-head fc09f2 single install

The steps are the same as the issue provides.

Results:
- the segmentation fault does not show up anymore
- the git Repo is processed and resources get created although not all of them are in the ready state 
![image](https://user-images.githubusercontent.com/6218999/103586707-dfb85480-4ea2-11eb-97ae-ef1dab3ae9bd.png)
 

I also tested with a modified version of our example git repo, and the git Repo is processed and app is deployed successfully 
![image](https://user-images.githubusercontent.com/6218999/103586804-17270100-4ea3-11eb-87f9-2e63e9a081d8.png)
